common/Perf/Azure.Test.Perf/EventPerfTest.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Test.Perf
+{
+    public abstract class EventPerfTest<TOptions> : PerfTest<TOptions> where TOptions : PerfOptions
```

Comment:
I continue to be opposed to this method of testing the processor.   I do not think that it provides any accurate representation of real-world performance nor that it is helpful as a synthetic benchmark.   The cost of acquiring the semaphore is going to dominate time spent in the event handler.   

We cannot treat a single dispatch to the event handler as an "operation" and consider it something that represents how the processor works.   Partitions are processed concurrently; this approach forces that into synchronization with no guarantees of fairness between partitions.   The result is either going to be that prefetch queues and the in-memory cache for each partition will fill up and remain full so we're essentially only testing the performance of a `foreach` loop and the semaphores, or we'll see a partition (or a couple of them) dominate and operate more normally while the others wait.    In the best case, results will be highly inconsistent for each run as the behavior will vary.

Respectfully, to reiterate what I've previously said - this performance framework is not suited to testing the processor due to it's assumption that everything can be segmented into a natural request/response cycle.  My concern is that we're introducing a test that offers no reflection of actual performance and that people will base decisions on.   

We should not add these tests.

Reply:
The new design of EventPerfTest should address this concern